include/sdk/heapCommon.hh

```diff
@@ -2,58 +2,53 @@
 
 #include <sdk/list.hh>
 
+#include <array>
 #include <cstring>
 
 namespace RVL
 {
 
-struct MEMiHeapHead
+class MEMiHeapHead
 {
-    u32 signature;
-    MEMLink link;
-    MEMList childList;
-    void *heapStart;
-    void *heapEnd;
-    u16 optFlag;
-};
+protected:
```

Comment:
In my opinion, I could get behind keeping the constructors always at the top regardless of access specifiers, but everything else (all the `fill` functions) should be moved to a new `protected` section below the `public` section.

Reply:
These are the only real methods that have any logic, hence why I chose to prioritize them.